### PR TITLE
cmake: fix Python install

### PR DESCRIPTION
**Pull request by jcarpent**
Required for Conda

### CODE REVIEW

**Comment by nim65s** [comment]
I think we should change `PYTHON_SITELIB` in the jrl-cmakemodule depending on `PROJECT_SITELIB_ROOT`. This would avoid the need for this kind of change here or in any other package.

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,20 @@
...
 
+IF(PROJECT_SITELIB_ROOT)
+  SET(${PY_NAME}_INSTALL_DIR ${PROJECT_SITELIB_ROOT}/${PYTHON_SITELIB}/${PY_NAME})
+ELSE()
+  SET(${PY_NAME}_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/${PYTHON_SITELIB}/${PY_NAME})
+ENDIF()
+
 FOREACH(python ${${PROJECT_NAME}_PYTHON_FILES})
-  PYTHON_INSTALL_ON_SITE(${PY_NAME} ${python})
+  INSTALL(FILES
```
Comment (nim65s) [inline python/CMakeLists.txt:15]: `PYTHON_INSTALL_ON_SITE` builds and install. This is useful to check the syntax and the build step, and to get a working python module in the build tree, which can be used by unit tests.

**Comment by wxmerkt** [comment]
I assume the `PYTHON_SITE_LIB` from upstream will still correctly support dist/site packages?

**Comment by nim65s** [comment]
I think that, if defined, `PROJECT_SITELIB_ROOT` should override `PYTHON_SITELIB`.  Otherwise, we should just keep `PYTHON_SITELIB` as it is now.

**Comment by jcarpent** [comment]
`PROJECT_SITELIB` would be ideal, to overwrite the `PYTHON_SITELIB`. Or we just need to allow this overwritting. @nim65s Could you handle it?

**Pull request closed without merging**